### DISCUSSION TITLE
Page transitions with framer motion doesn't work as expected

### DISCUSSION THREAD

**Discussion by adrifer** [remix-run/remix#3685] [Q&A]
Hello everyone!

This is my first time with Remix and also with Framer Motion so probably this is some silly question. I've tried to reproduce the [Framer-route-animation](https://github.com/remix-run/remix/tree/main/examples/framer-route-animation) example in a fresh indie-stack new project. But I'm not getting the same behavior as in the example. In my test I'm seeing how the new route is rendered before doing the page transition (with the new route) and then the enter animation with the same content. 

This is a gif with the result: 
![PageTransitions](https://user-images.githubusercontent.com/812246/178082394-818febf5-4ca7-410f-a65f-d2fdf7230975.gif)

As you can see the new route is rendered before doing the page transition. When I'm in the second note and I click in the first page, I can see how "Second note" changes to "First note" and then I see the transition.

The only thing I did to do this is to install framer-motion and then change the App component with this:
```
  export default function App() {
  const outlet = useOutlet();
  return (
    <html lang="en" className="h-full">
      <head>
        <Meta />
        <Links />
      </head>
      <body className="h-full">
        <AnimatePresence exitBeforeEnter initial={false}>
          <motion.main
            key={useLocation().key}
            initial={{ y: "40%", opacity: 0 }}
            animate={{ y: "0", opacity: 1 }}
            exit={{ y: "-40%", opacity: 0 }}
            transition={{ duration: 1 }}
          >
            {outlet}
          </motion.main>
        </AnimatePresence>
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
```

Please let me know if you know what I'm doing wrong (probably I'm missing something and it's a silly thing, but I didn't find it so far).

Thank you in advance!

**Comment by adrifer** [comment]
Can anyone help me with this?

**Comment by kwiat1990** [comment]
I seems like I also struggle with this one. There's no information about animating routes in Remix docs and the example repo also doesn't work for me. I have installed `react-router-dom` but as soon as I change a route I get errors from `useLoaderData`. I use Remix `1.6.7`.
